worker/worker.go

```diff
@@ -45,8 +45,12 @@ func (w *Pool) StartWorkers() {
 				//@TODO - Should add integration tests to prove that the worker receives the same message that it produced, on the delivery channel it created
 
 				err := w.kafkaProducer.ProduceBulk(request.GetEvents(), request.ConnectionIdentifier.Group, deliveryChan)
-				totalErr := 0
 
+				if request.AckFunc != nil {
+					request.AckFunc(err)
+				}
+
+				totalErr := 0
 				if err != nil {
 					for _, err := range err.(publisher.BulkError).Errors {
```

Comment:
(suggestion): we can call the `BulkError.Error()` to log the message instead of looping over the errors
https://github.com/odpf/raccoon/blob/main/publisher/kafka.go#L148

Reply:
I believe the above method should also be fixed because it uses `fmt.Sprintf` which will cause performance problems.